src/transformStateWithClones.js

```diff
@@ -8,6 +8,78 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const array = [];
+  const object = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(object, action.extraData);
+
+        const copy = { ...object };
+
+        array.push(copy);
+
+        break;
+      case 'removeProperties':
+        const removeArray = action.keysToRemove;
+
+        for (const key of removeArray) {
+          if (key in object) {
+            delete object[key];
+          }
+        }
+
+        const copy2 = { ...object };
+
+        array.push(copy2);
+
+        break;
+      case 'clear':
+        for (const key in object) {
+          delete object[key];
+        }
+
+        array.push({});
```

Comment:
If you perform same action in all switch cases - do it just once afterwards.

you push copy in each case, so don't repeat yourself, move it after switch case, please

checklist 4 https://github.com/mate-academy/js_stateful-clones/blob/master/checklist.md